### DISCUSSION TITLE
version diffing only by git tag

### DISCUSSION THREAD

**Discussion by ma-multipla** [lerna/lerna#3447] [Q&A]
Is there a possibility to omit having to commit the version changes in the package.json? We have a job that can be triggered manually in a merge request to create a prerelease of the branch following conventional commits. Lerna updates the package.json of the affected packages and pushes them in the registry with `lerna publish`. We are currently amending the latest commit and force pushing it to the branch but each developer needs to remember to pull after creating a prerelease or it will diverge. Is there a possibility not to have to commit the version changes and just temporarily change the version for the publish but then omit it and only do the version diffing via git tags?

**Comment by ghiscoding** [comment]
maybe [--no-git-tag-version](https://github.com/lerna/lerna/tree/main/commands/version#--no-git-tag-version) or [--no-push](https://github.com/lerna/lerna/tree/main/commands/version#--no-push) option.

**Reply by ma-multipla** [reply]
> maybe [--no-git-tag-version](https://github.com/lerna/lerna/tree/main/commands/version#--no-git-tag-version) or [--no-push](https://github.com/lerna/lerna/tree/main/commands/version#--no-push) option.

Sadly those options don't solve our problems `--no-git-tag-version` also doesn't tag the release and then there is probably no way for lerna to track that a prerelease has already been created as we also don't want to commit and push the version changes in the package.json with this flag. We also need to maintain the possibility to create multiple prereleases this requires tracking previous prerelease, just without having to commit the version changes if possible.
